security-admin/src/main/java/org/apache/ranger/rest/GdsREST.java

```diff
@@ -1911,18 +1912,23 @@ List<RangerGrant> transformPolicyItemsToGrants(List<RangerPolicyItem> policyItem
             List<RangerPolicyItemCondition> policyItemConditions = policyItem.getConditions();
 
             List<String> policyItemAccessTypes     = policyItemAccesses.stream().map(RangerPolicyItemAccess::getType).collect(Collectors.toList());
-            List<String> policyItemConditionValues = policyItemConditions.stream().flatMap(x -> x.getValues().stream()).collect(Collectors.toList());
+
+            List<RangerPolicy.RangerPolicyItemCondition> policyItemConditionExpressions = policyItemConditions.stream().filter(c -> c.getType().equals(GDS_POLICY_EXPR_CONDITION)).collect(Collectors.toList());
+            List<String> expressions = policyItemConditionExpressions.stream().flatMap(x -> x.getValues().stream()).collect(Collectors.toList());
+
+            List<RangerPolicy.RangerPolicyItemCondition> policyItemConditionValiditySchedules = policyItemConditions.stream().filter(c -> c.getType().equals(GDS_POLICY_VALIDITY_SCHEDULE_CONDITION)).collect(Collectors.toList());
```

Comment:
Consider avoiding creation of lists `policyItemConditionExpressions` and `policyItemConditionValiditySchedules` with the following approach:

```
List<String> expressions       = new ArrayList<>();
List<String> validitySchedules = new ArrayList<>();

if (CollectionUtils.isNotEmpty(policyItemConditions)) {
    policyItemConditions.forEach(condition -> {
        if (condition.getType().equals(GDS_POLICY_EXPR_CONDITION)) {
            expressions.addAll(condition.getValues());
        } else if (condition.getType().equals(GDS_POLICY_VALIDITY_SCHEDULE_CONDITION)) {
            validitySchedules.addAll(condition.getValues());
        }
    });
}
```

Also, please update the description for changes in RangerGrant.

Reply:
@mneethiraj  This review comment has been addressed and new update the pull request done. Thanks.